chain/sync_manager.go

```diff
@@ -250,10 +622,6 @@ func (sbs *syncBucketSet) Empty() bool {
 	return len(sbs.buckets) == 0
 }
 
-type syncTargetBucket struct {
-	tips []*types.TipSet
-}
-
 func (stb *syncTargetBucket) sameChainAs(ts *types.TipSet) bool {
```

Comment:
This is asserting that the tipset has a direct parent/child connection with *any* (but not all) of the tipsets in this bucket. Later `(*syncBucketSet).Insert()` will add this tipset to the bucket. If the bucket contains a parent/child tipsets `A <- B` and then comes tipset `C` also child of `A` but unrelated to `B` we would accept it just the same leading to a bucket with two tipsets from potentially different chains:
```
A <- B
A <- C
```